### PR TITLE
Fix doxygen comments for Operation<T>

### PR DESCRIPTION
**Pull request by RickWinter**

### CODE REVIEW

File: sdk/core/azure-core/inc/azure/core/operation_status.hpp [hunk 1]
```diff
@@ -24,28 +24,28 @@ namespace Azure { namespace Core {
...
      *
      * @param value A non-absent value to initialize with.
      */
     explicit OperationStatus(const std::string& value) : m_value(value) {}
     /**
-     * @brief Construct a #OperationStatus with \p value.
+     * @brief Construct an #OperationStatus with \p value.
      *
-     * @param value A non-absent value to initialize with.
+     * @param value The value to initialize with.
```
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/operation_status.hpp:35]: Make a similar fix for the two other ctors above:
https://github.com/Azure/azure-sdk-for-cpp/blob/cb32cbeeb942d3dd3f1d3490febf68c76206ff0c/sdk/core/azure-core/inc/azure/core/operation_status.hpp#L29-L41

File: sdk/core/azure-core/inc/azure/core/operation_status.hpp [hunk 2]
```diff
@@ -24,28 +24,28 @@ namespace Azure { namespace Core {
 
   public:
     /**
-     * @brief Construct a #OperationStatus with \p value.
+     * @brief Construct an #OperationStatus with \p value.
      *
      * @param value A non-absent value to initialize with.
```
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/operation_status.hpp:29]: Apply the same fix to other uses of `non-absent`.
```suggestion
     * @param value The value to initialize with.
```

File: sdk/core/azure-core/test/ut/operation_status.cpp [hunk 3]
```diff
@@ -48,4 +48,9 @@ TEST(OperationStatus, Custom)
   OperationStatus status4 = OperationStatus(std::string("CustomValue"));
   EXPECT_EQ(status4.Get(), "CustomValue");
   EXPECT_NE(status4, OperationStatus::NotStarted);
+
+  EXPECT_EQ(status1, status2);
+  EXPECT_EQ(status2, status3);
+  EXPECT_EQ(status3, status4);
+
```
Comment (ahsonkhan) [inline sdk/core/azure-core/test/ut/operation_status.cpp:55]: Nice :)

**Review by ahsonkhan — APPROVED** [review]
Other than the `non-absent` fix applied everywhere, looks good.

File: sdk/core/azure-core/inc/azure/core/operation_status.hpp (hunk 1 above)
Comment (RickWinter) [reply sdk/core/azure-core/inc/azure/core/operation_status.hpp:35]: good catch, fixed

**Review by ahsonkhan — APPROVED** [review]

**Comment by ahsonkhan** [comment]
This is a follow up to https://github.com/Azure/azure-sdk-for-cpp/pull/1186

**Pull request merged**